### PR TITLE
Ensure empty queries are returned as invalid (#33095)

### PR DESCRIPTION
**Pull request by cbismuth**
Null queries were causing an `ActionRequestValidationException`. To work around this issue, null queries are reset to a "match all" query which was already the default value.

Empty and whitespaces-only queries were already passing the test suite. Specific test cases have just been added to ensure that.

### CODE REVIEW

**Review by javanna — COMMENTED** [review]
thanks a lot for your contribution @cbismuth ! I have reviewed it, left a couple of comments, and also realized that the original bug is not present anymore, so the PR is not really fixing it. That said I am not sure we should incorporate this fix given that it was submitted to fix an issue that is already fixed.

**Comment by javanna** [comment]
One more thing, I am not sure how this change relates to the high-level REST client, can you clarify that? Maybe I am missing something. Thanks!

File: client/rest-high-level/src/test/java/org/elasticsearch/client/IndicesClientIT.java [hunk 1]
```diff
@@ -1267,13 +1268,21 @@ public void testPutTemplateBadRequests() throws Exception {
         assertThat(unknownSettingError.getDetailedMessage(), containsString("unknown setting [index.this-setting-does-not-exist]"));
     }
 
-    public void testValidateQuery() throws IOException{
+    public void testValidateQueryWithNullQuery() throws IOException {
```
[the remark this answers is not in the document]
Comment (cbismuth) [reply client/rest-high-level/src/test/java/org/elasticsearch/client/IndicesClientIT.java]: Yes, I'll add a such a test.

**Comment by cbismuth** [comment]
You're welcome @javanna.

I've added tests in the high-level REST client, that's why I've set this PR title. But your're right, the NPE check is in the server/action module.

I'll develop a cURL recreation script and use a debugger to have a better understanding of how validation works. Then, I'll check if a REST integration test has been developed and add a new one if none exists.

I'll let you know.

File: client/rest-high-level/src/test/java/org/elasticsearch/client/IndicesClientIT.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (cbismuth) [reply client/rest-high-level/src/test/java/org/elasticsearch/client/IndicesClientIT.java]: @javanna, after reviewing my own commit, I now realize why we didn't understand each other. I've read the issue description far to fast, I though it was written: "an empty query should be marked as valid" ... my bad, sorry for all that noise, these are my first baby steps as an Elasticsearch contributor.

[the remark this answers is not in the document]
Comment (javanna) [reply client/rest-high-level/src/test/java/org/elasticsearch/client/IndicesClientIT.java]: no worries

**Pull request closed without merging**